pkgs/development/libraries/libidn2/default.nix

```diff
@@ -20,9 +22,15 @@ stdenv.mkDerivation rec {
 
   patches = optional stdenv.isDarwin ./fix-error-darwin.patch;
 
+  # hack found here: https://github.com/sethrj/spack/commit/93393fe2f510a8ed9ace65534ee80e5433e19943
+  # works around the fact that autoreconf'ing libidn2 causes it to insist on gtk-doc being present
+  GTKDOCIZE = if stdenv.hostPlatform != stdenv.buildPlatform then "echo" else null;
+
   # The above patch causes the documentation to be regenerated, so the
   # documentation tools are required.
-  nativeBuildInputs = optionals stdenv.isDarwin [ help2man texinfo ];
+  nativeBuildInputs = optionals stdenv.isDarwin [ help2man texinfo ]
+  # libidn2's Makefile "helpfully" forces an autoreconf if it notices changes to the ./configure script
+  ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [ autoreconfHook buildPackages.pkgconfig];
```

Comment:
That is pretty standard across many projects. Also pkgconfig should be pkg-config.